### PR TITLE
FUSETOOLS-2004 - new attribute for SOAP message composer

### PR DESCRIPTION
**Pull request by bfitzpat**
Updated EMF model and added new checkbox to SOAP service and reference
bindings.

### CODE REVIEW

**Comment by bfitzpat** [comment]
@apupier Can you give this a quick look when you get a chance? Ultimately it's just an update to the EMF model for SwitchYard we have to do when the runtime schemas are updated - and then I just added a new control to the corresponding UI bits.

**Comment by apupier** [comment]
testPR

File: eclipse/plugins/org.switchyard.tools.models.switchyard1_0/src/org/switchyard/tools/models/switchyard1_0/soap/impl/MessageComposerTypeImpl.java [hunk 1]
```diff
@@ -187,7 +273,9 @@ public String toString() {
         if (eIsProxy()) return super.toString();
 
         StringBuffer result = new StringBuffer(super.toString());
-        result.append(" (unwrapped: ");
+        result.append(" (copyNamespaces: ");
+        if (copyNamespacesESet) result.append(copyNamespaces); else result.append("<unset>");
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.models.switchyard1_0/src/org/switchyard/tools/models/switchyard1_0/soap/impl/MessageComposerTypeImpl.java]: I really prefer when there is a single statement per line
If on one line, for very simple case, I prefer using the copyNamespacesESet ? result.append(copyNamespaces) : result.append("<unset>") notation

File: eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingReferenceComposite.java [hunk 2]
```diff
@@ -202,6 +203,11 @@ public void widgetSelected(final SelectionEvent e) {
         upChxGD.horizontalSpan = 3;
         _unwrappedPayloadCheckbox.setLayoutData(upChxGD);
 
+        _copyNamespacesCheckbox = createCheckbox(composite, "Copy Namespaces");
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingReferenceComposite.java]: You don't externalize Strings in SwitchYard?

File: eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingReferenceComposite.java [hunk 3]
```diff
@@ -202,6 +203,11 @@ public void widgetSelected(final SelectionEvent e) {
...
+        GridData cnChxGD = new GridData(GridData.FILL_HORIZONTAL);
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingReferenceComposite.java]: this is fine like that too but I would suggest to use GridDataFactory.fillDefaults().span(3,1).create();
I like the usage of the Factory which is providing a Fluent API

File: eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingReferenceComposite.java [hunk 4]
```diff
@@ -423,21 +429,36 @@ private void bindControls(final DataBindingContext context) {
...
 
         binding = context
                 .bindValue(
                         SWTObservables.observeSelection(_unwrappedPayloadCheckbox), msgComposerUnwrappedValue,
                         new EMFUpdateValueStrategyNullForEmptyString(null, UpdateValueStrategy.POLICY_CONVERT), null);
         ControlDecorationSupport.create(SWTValueUpdater.attach(binding), SWT.TOP | SWT.LEFT);
 
+        org.eclipse.core.databinding.Binding cnBinding = context
+                .bindValue(
+                        SWTObservables.observeSelection(_copyNamespacesCheckbox), msgComposerCopyNamespacesValue,
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingReferenceComposite.java]: You should consider using WidgetProperties for databinding which is the new API (for few years now)
(not blocking for the PR, just to think about it next time yuou manipulate Databinding)

File: eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingServiceComposite.java [hunk 5]
```diff
@@ -220,6 +221,11 @@ public void widgetSelected(final SelectionEvent e) {
         GridData upChxGD = new GridData(GridData.FILL_HORIZONTAL);
         upChxGD.horizontalSpan = 3;
         _unwrappedPayloadCheckbox.setLayoutData(upChxGD);
+        
+        _copyNamespacesCheckbox = createCheckbox(composite, "Copy Namespaces");
+        GridData cnChxGD = new GridData(GridData.FILL_HORIZONTAL);
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingServiceComposite.java]: I don't see differences in SOAPBindingServiceComposite compared to SOAPBindingReferenceComposite (at least for the code added)
Can't we factorize the code?

**Comment by bfitzpat** [comment]
Thanks for the feedback Aurelien - I'll see what I can do to address it today. :)

File: eclipse/plugins/org.switchyard.tools.models.switchyard1_0/src/org/switchyard/tools/models/switchyard1_0/soap/impl/MessageComposerTypeImpl.java (hunk 1 above)
Comment (bfitzpat) [inline eclipse/plugins/org.switchyard.tools.models.switchyard1_0/src/org/switchyard/tools/models/switchyard1_0/soap/impl/MessageComposerTypeImpl.java]: This is all generated code, so I'm not going to mess with it at the risk of screwing something up in EMF, but I think I understand what you're after.

File: eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingReferenceComposite.java (hunk 2 above)
Comment (bfitzpat) [inline eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingReferenceComposite.java]: Yes - will fix this.

File: eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingReferenceComposite.java (hunk 3 above)
Comment (bfitzpat) [inline eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/soap/SOAPBindingReferenceComposite.java]: I prefer breaking it out into more readable lines myself, but I've seen the form you are suggesting. John uses it. Is it more efficient? Or is this merely a question of style?

File: eclipse/plugins/org.switchyard.tools.models.switchyard1_0/src/org/switchyard/tools/models/switchyard1_0/soap/impl/MessageComposerTypeImpl.java (hunk 1 above)
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.models.switchyard1_0/src/org/switchyard/tools/models/switchyard1_0/soap/impl/MessageComposerTypeImpl.java]: ok I didn't seen that it was generated code here.

**Pull request merged**